src/include/shared/sgxlkl_enclave_config_gen.h

```diff
@@ -0,0 +1,148 @@
+#ifndef SGXLKL_ENCLAVE_CONFIG_GEN_H
+#define SGXLKL_ENCLAVE_CONFIG_GEN_H
+
+/* Automatically generated from ../../tools/schemas/enclave-config.schema.json; do not modify. */
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <elf.h>
+
+#define SGXLKL_ENCLAVE_CONFIG_VERSION 1UL
+
+typedef enum
+{
+    UNKNOWN_MODE = 0,
+    SW_DEBUG_MODE = 1,
+    HW_DEBUG_MODE = 2,
+    HW_RELEASE_MODE = 3
+} sgxlkl_enclave_mode_t;
+
+const char* sgxlkl_enclave_mode_t_to_string(sgxlkl_enclave_mode_t e);
+sgxlkl_enclave_mode_t string_to_sgxlkl_enclave_mode_t(const char *e);
+
+typedef enum
+{
+    ENCLAVE_MMAP_FILES_NONE = 0,
+    ENCLAVE_MMAP_FILES_PRIVATE = 1,
+    ENCLAVE_MMAP_FILES_SHARED = 2
+} sgxlkl_enclave_mmap_files_t;
+
+const char* sgxlkl_enclave_mmap_files_t_to_string(sgxlkl_enclave_mmap_files_t e);
+sgxlkl_enclave_mmap_files_t string_to_sgxlkl_enclave_mmap_files_t(const char *e);
+
+typedef enum
+{
+    EXIT_STATUS_FULL = 0,
+    EXIT_STATUS_BINARY = 1,
+    EXIT_STATUS_NONE = 2
+} sgxlkl_exit_status_mode_t;
+
+const char* sgxlkl_exit_status_mode_t_to_string(sgxlkl_exit_status_mode_t e);
+sgxlkl_exit_status_mode_t string_to_sgxlkl_exit_status_mode_t(const char *e);
+
+typedef struct sgxlkl_clock_res_config
+{
+    char resolution[17];
+} sgxlkl_clock_res_config_t;
+
+typedef struct sgxlkl_enclave_mount_config
+{
+    bool create;
+    char destination[256];
+    size_t key_len;
+    uint8_t* key;
+    char* key_id;
+    bool fresh_key;
+    bool readonly;
+    char* roothash;
+    size_t roothash_offset;
+    size_t size;
+} sgxlkl_enclave_mount_config_t;
+
+typedef struct sgxlkl_enclave_root_config
+{
+    size_t key_len;
+    uint8_t* key;
+    char* key_id;
+    char* roothash;
+    size_t roothash_offset;
+    bool readonly;
+    bool overlay;
+} sgxlkl_enclave_root_config_t;
+
+typedef struct sgxlkl_enclave_wg_peer_config
+{
+    char* key;
+    char* allowed_ips;
+    char* endpoint;
+} sgxlkl_enclave_wg_peer_config_t;
+
+typedef struct sgxlkl_enclave_wg_config
+{
+    char* ip;
+    uint32_t listen_port;
+    char* key;
+    size_t num_peers;
+    sgxlkl_enclave_wg_peer_config_t* peers;
+} sgxlkl_enclave_wg_config_t;
+
+typedef struct sgxlkl_image_sizes_config
+{
+    uint64_t num_heap_pages;
+    uint64_t num_stack_pages;
+} sgxlkl_image_sizes_config_t;
+
+typedef struct sgxlkl_enclave_config
+{
+    sgxlkl_enclave_mode_t mode;
+    char* net_ip4;
+    char* net_gw4;
+    char* net_mask4;
+    char hostname[32];
```

Comment:
32 characters seems very small for a fully-qualified Azure hostname.

Reply:
Sure, pick a number then, this is what it currently is.

Reply:
`char*`?

Reply:
Sure, can do! Since it's been given a very specific limit, I assumed this came from somewhere else. If you can think of a place that might rely on this please let me know.

Reply:
@prp, do you know if this is a limitation from somewhere specific, or just a size that was picked?

Looking at some of the Linux docs, it appears that Linux has 64- and 255-byte limits for host names in different contexts.

Reply:
I don't have an explanation for this. I think that the size should be increased.

Reply:
Done